adapta/logs/_internal_logger.py

```diff
@@ -0,0 +1,178 @@
+"""
+ Shared functionality for the MetadataLogger enricher implementations.
+"""
+#  Copyright (c) 2023. ECCO Sneaks & Data
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+
+import logging
+from typing import Optional, Dict
+
+from adapta.logs._internal import MetadataLogger
+
+
+class _InternalLogger:
```

Comment:
You are creating an internal class with public methods. These methods are exposed through inherited classes `AsyncLogger` and `SemanticLogger`, which can be confusing.

I would suggest making the `meta_*` methods internal and annotate them with the `@final` decorator to prevent overriding issues.

However, it might be even better if you use composition instead of inheritance. I don't see any reason for inheritance in this case.

Reply:
will reach out on Monday re improving this then :)